src/scripts/build_nst_da.py

```diff
@@ -0,0 +1,305 @@
+            file_size = audio_file.stat().st_size
+            if file_size < 8192:
+                logger.info(
+                    f"Removing {audio_file.name!r} as its size {file_size} bytes"
+                    " is too small."
+                )
+                audio_file.unlink()
+
+
+def get_suffix(string: str | Path) -> str:
+    """Get the suffix of a string.
+
+    Contrary to Path.suffix, this also works for strings with multiple suffixes.
+
+    Args:
+        string: The string to get the suffix from.
+
+    Returns:
+        The suffix of the string, or an empty string if there is no suffix.
+    """
+    string = str(string)
+    matches = re.search(r"(\.[^.]+)+$", string.split("/")[-1])
+    if matches:
+        return matches.group()
+    else:
+        return ""
+
+
+def build_huggingface_dataset() -> DatasetDict:
+    """Sets up the metadata files and builds the Hugging Face dataset.
+
+    Returns:
+        The Hugging Face dataset.
+    """
+    rng = np.random.default_rng(seed=4242)
+
+    def ensure_int(value: int | str | None) -> int | None:
+        if value is None:
+            return None
+        try:
+            return int(value)
+        except ValueError:
+            return None
+
+    def fix_text_column(text: str) -> str:
+        if text == "( ... tavshed under denne indspilning ...)":
+            return ""
+        return text
+
+    columns_to_keep = {
+        "filename_both_channels": "audio",
+        "text": "text",
+        "Speaker_ID": "speaker_id",
+        "Age": "age",
+        "Sex": "sex",
+        "Region_of_Dialect": "dialect",
+        "RecDate": "recording_date",
+        "RecTime": "recording_time",
+    }
+
+    dataset_dict: dict[str, Dataset] = dict()
+    for split in ["train", "test"]:
+        metadata_path = Path(split) / "metadata.csv"
+        metadata_df = pd.read_csv(metadata_path, low_memory=False)
+        metadata_df = metadata_df[columns_to_keep.keys()]
+        metadata_df = metadata_df.rename(columns=columns_to_keep)
+        metadata_df.age = metadata_df.age.map(ensure_int)
+        metadata_df.speaker_id = metadata_df.age.map(ensure_int)
+        metadata_df.text = metadata_df.text.map(fix_text_column)
+        metadata_df = metadata_df.dropna()
+        metadata_df = metadata_df.convert_dtypes()
+
+        audio_dir = Path(split) / "audio"
+        audio_filenames = pd.Series(
+            [str(audio_file) for audio_file in audio_dir.glob("*.wav")]
+        )
+        recording_datetimes: list[str] = list()
+        for idx, row in tqdm(
+            iterable=metadata_df.iterrows(),
+            total=len(metadata_df),
+            desc=f"Extracting file names for the {split} split",
+        ):
+            datetime = dt.datetime.strptime(
+                f"{row.recording_date}T{row.recording_time}", "%d %b %YT%H:%M:%S"
+            )
+            original_filename = (
+                row.audio.lower().split("-")[-1].split("_")[-1].replace(".wav", "")
```

Comment:
This is quite unclear to me. Might be obvious when looking at examples of filenames, but maybe a comment with an example?

Reply:
Added some more descriptive comments now